core/src/main/java/org/apache/calcite/schema/lookup/IgnoreCaseLookup.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to you under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.calcite.schema.lookup;
+
+import org.apache.calcite.util.LazyReference;
+import org.apache.calcite.util.NameMap;
+
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * An abstract base class for lookups implementing caseinsensitive lookup.
+ *
+ * @param <T> Element type
+ */
+public abstract class IgnoreCaseLookup<T> implements Lookup<T> {
+
+  /**
+   * This member is used to lazily load the list of all names into memory.
+   *
+   * <p>A {@link NameMap} is used, which is capable to lookup names in a
+   * caseinsensitive way.
+   */
+  private LazyReference<NameMap<String>> nameMap = new LazyReference<>();
+
+  /**
+   * Returns a named entity with a given name, or null if not found.
+   *
+   * @param name Name
```

Comment:
I don't insist having documentation for parameters which are "obvious" - here the documentation does not add much. Some tools do. For `@return` it's use to say "Entity with the specified name, or null when such entity is not found".

Reply:
OK